org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -26,18 +26,28 @@ class Translator {
     /**
      * Get the hub definition and devices
      */
-    get(expand, payload) {
-        return this.getPlatforms(expand, payload);
+    get(expand, payload, verification) {
+        return this.getPlatforms(expand, payload, verification);
     }
 
     /**
      * Get the list of devices discovered through the hub.
      */
-    getPlatforms(expand, payload) {
+    getPlatforms(expand, payload, verification) {
 
         // Payload can contain one or more platforms defined using the provider schema.  This should return those platforms
         // converted to the opent2t/ocf representation.
         if (payload !== undefined) {
+            // Callculate the HMAC for the payload using the secret
+            if (verification !== undefined && verification.key !== undefined) {
+                // use secret to calculate HMAC
+                var hmac = require('crypto').createHmac('sha1', verification.key).update(payload).digest('hex');
```

Comment:
this feels like a long line of code, can you at the very least break out the require call?